### ISSUE TITLE
function for creating remake recipes in task steps 

### ISSUE THREAD

**Issue by jzwart** [USGS-R/scipiper#56]
turn this into a scipiper function 
 ``` # function to sprintf a bunch of key-value (string-variableVector) pairs, then
  # paste them together with a good separator for constructing remake recipes
  psprintf <- function(..., sep='\n      ') {
    args <- list(...)
    strs <- mapply(function(string, variables) {
      spargs <- if(string == '') list(variables) else c(list(string), as.list(variables))
      do.call(sprintf, spargs)
    }, string=names(args), variables=args)
    paste(strs, collapse=sep)
  }
```
to be used in creating tasks steps and function calls like so: 
```
  gif_frame <- scipiper::create_task_step(
    step_name = 'gif_frame',
    target_name = function(task_name, step_name, ...){
      file.path(folders$tmp, sprintf('gif_frame_%s.png', task_name))
    },
    command = function(task_name, ...){
      cur_task <- dplyr::filter(rename(tasks, tn=task_name), tn==task_name)
      psprintf(
        "create_storm_frame(",
        "png_file=target_name",
        "config='%s',"=cfg,
        "'%s',"=view_ind,
        "'%s',"=basemap_ind,
        "'%s'"=storm_line_ind,
        "5_vizprep/out/storm_point_%s.rds.ind,"=cur_task$date_hour,
        "5_vizprep/out/precip_raster_%s.rds.ind,"=cur_task$date_hour,
        "5_vizprep/out/streamdata_%s.rds.ind,"=cur_task$date_hour,
        sep="\n      ")
    }
  )
```

**Comment by aappling-usgs** [comment]
I think we could add this function pretty much as-is.

**Comment by aappling-usgs** [comment]
That said, this newer version might be better? Figure out what it does and whether it does it better: https://github.com/USGS-VIZLAB/gage-conditions-gif/blob/befeb5f535f5cde597c47e8d800c8da6f4eccad6/6_visualize/src/create_gif_tasks.R#L87-L106

**Comment by aappling-usgs** [comment]
Resolved by #90
